jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -16,152 +16,227 @@
-                            removeGroupFromRole(group, crs.getName());
-                        }
-                    }));
-                }
-        );
+            @Override
+            public void execute() {
+                editRoleAssignmentView.setValue(caseRoleAssignmentSummary);
+                editRoleAssignmentView.show(() -> assignToRole(oldUserAssignments, oldGroupsAssignments));
+            }
+        };
     }
 
-    protected void addAssignment(final String userName,final String groupName, final String roleName) {
-        if(!isNullOrEmpty(userName) && !isNullOrEmpty(groupName)){
-            caseService.call(
-                    (Void) -> findCaseInstance()
-            ).assignGroupAndUserToRole(serverTemplateId, containerId, caseId, roleName, userName,groupName);
-        }else {
-            if (!isNullOrEmpty(userName)) {
-                addUserToRole(userName, roleName);
+    private CaseRoleAction getRemoveAction(String roleName, boolean enabledAction,
+                                           List<String> oldUserAssignments,
+                                           List<String> oldGroupsAssignments) {
+        return new CaseRoleAction() {
+            @Override
+            public String label() {
+                return translationService.format(REMOVE_ALL_ASSIGNMENTS);
             }
-            if(!isNullOrEmpty(groupName)){
-                addGroupToRole(groupName,roleName);
+
+            @Override
+            public boolean isEnabled() {
+                return enabledAction;
+            }
+
+            @Override
+            public void execute() {
+                confirmPopup.show(translationService.format(REMOVE_ASSIGNMENT),
+                        translationService.format(REMOVE),
+                        translationService.format(REMOVE_ALL_USERS_GROUPS_FROM_ROLE, roleName),
+                        () -> storeRoleAssignments(roleName, Collections.emptyList(),
+                                Collections.emptyList(), oldUserAssignments, oldGroupsAssignments));
             }
+        };
+    }
+
+    protected CaseRoleAssignmentSummary getRoleAssignment(final CaseInstanceSummary cis, final String roleName) {
+        if (cis.getRoleAssignments() != null && cis.getRoleAssignments().size() > 0) {
+            return cis.getRoleAssignments()
+                    .stream()
+                    .filter(ra -> roleName.equals(ra.getName()))
+                    .findFirst().orElse(CaseRoleAssignmentSummary.builder().name(roleName).build());
         }
+        return CaseRoleAssignmentSummary.builder().name(roleName).build();
     }
 
-    protected void addUserToRole(final String userName, final String roleName) {
-        caseService.call(
-                (Void) -> findCaseInstance()
-        ).assignUserToRole(serverTemplateId, containerId, caseId, roleName, userName);
+    protected List<String> validateRolesAssignments(final CaseDefinitionSummary caseDefinition,
```

Comment:
Would be nice if we could share this code between both presenter classes, NewCaseInstancePresenter and this one.